committee-steering/sig-governance.md

```diff
@@ -0,0 +1,629 @@
+# SIG Governance Template
+
+## Goals
+
+The following document outlines a template governance structure for
+SIGs to follow.  Its goals are to provide the necessary structure
+for SIGs to self organize and self govern.
+
+This document is focused on:
+
+- Outlining organizational responsibilities
+- Outlining organizational roles
+- Outlining processes and tools
+
+Specific attention has been given to:
+
+- The role of technical leadership
+- The role of project leadership
+- The process for making technical decisions
+- The process for ensuring continued test health
+
+## How to use this template
+
+SIGs are diverse in nature, and may have different goals based
+on what areas of the project the SIG touches.  Accordingly SIGs will
+need to adapt this template to their specific needs.  This may mean
+removing or changing roles, responsibilities, and processes.  When
+diverging from this template, please help us understand why in the
+PR description so we can improve the process and iterate on this
+document.
+
+**Note:** There are 2 charter template versions:
+
+- a super-lite TL;DR version that only provides a abstract structure.
+- a detailed version that attempts to concretely define a high
+  degree of structure.
+
+SIGs may either choose to start with the TL;DR charter and add
+structure from the detailed version, or to start with the detailed
+version and remove structure that does not provide direct value.
+
+**Note:** Specific responsibilities towards the project are required
+for some SIGs and should be included in the charter - e.g. ensuring
+test health.
+
+FAQ:
+
+- [Why is it so long?](#why-is-it-so-long)
+- [Why so much structure?](#why-so-much-structure)
+- [Why use mailing lists for tracking roles?](#why-use-mailing-lists-for-tracking-roles)
+- [Why track membership as a special role?](#why-track-membership-as-a-special-role)
+- [How do the results from the survey fit in?](#how-do-the-results-from-the-survey-fit-in)
+
+---
+
+# SIG ~~Foo~~ Governance TL;DR Version
+
+## Roles
+
+- Project Leads
+  - Run operations and processes governing the SIG
+  - Outgoing Project Leads may propose new ones -
+  - Number: 2-3
+- Technical Leads
```

Comment:
This needs to be reconciled with subprojects and owners.

https://github.com/kubernetes/community/issues/1673

Reply:
How do you propose that should happen? or are you suggesting that we just add a comment here that the TLs should be reconciled with subprojects and owners in a way the SIG deems appropriate?

Reply:
Different sub-projects have different owners in the form of owners/maintainers (see the [membership doc](https://github.com/kubernetes/community/blob/master/community-membership.md)). Wouldn't the owners/maintainers for a sub-project under the SIG be technical leads for that sub-project (or part of it)?

I would use SIG Apps and Helm as a worse case test when you include everything on the kubernetes-helm org for any approach. It's sufficiently complicated enough to do edge case testing. Though, that may not apply to the simple template.

Reply:
Yes, owners are intended to be the technical leads of a particular subproject, under a particular SIG:
https://github.com/kubernetes/community/issues/1673

It's already a rung on the contributor ladder, and I intend to fold maintainers into owners:
https://github.com/kubernetes/community/blob/master/community-membership.md#owner

What I'm not sure about is whether we need overall technical leaders for SIGs. The owners of all SIG subprojects could collectively serve as the technical leadership for the SIG. Some SIGs do have de facto overall leads. I think it's an open question whether it makes sense for every SIG, whether we want to make that pattern official, or whether it would be materially different than the union of subproject owners. Maybe each SIG could have a Technical Steering Committee made up of its subproject owners? Or we could recommend both patterns (uber owners and owners committees) as alternatives.